### PR TITLE
Bubble sort merge request

### PR DESCRIPTION
**Pull request by Mikearaya**
**On this pull request, we have created an optimized  bubble algorithm .**
- The algorithm sorts array items in an ascending order

## We kindly request your review so we can merge to master branch :bow: 

**Thank you**

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      end
+      break if j == length - 1
+
+      j += 1
+    end
+    break if is_sorted
+
+    i += 1
+  end
+  array.to_s
```
Comment (mosesogwo) [inline bubble_sort.rb]: The return value should be the sorted array and not a string

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      if yield(array[i], array[i + 1]).positive?
+        array[i], array[i + 1] = array[i + 1], array[i]
+        is_sorted = false
+      end
+    end
+    break if is_sorted
+
+    index += 1
+  end
+  array.to_s
```
Comment (mosesogwo) [inline bubble_sort.rb]: The return value should be the sorted array and not a string

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    end
+    break if is_sorted
+
+    i += 1
+  end
+  array.to_s
+end
+
+def bubble_sort_by(array)
+  return 'No block given' unless block_given?
```
Comment (mosesogwo) [inline bubble_sort.rb]: Ruby has a default error that is raised when an expected block is not provided. Let's remove this error handling and allow the default error.

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+def bubble_sort(array)
+  is_sorted = false
+
```
Comment (mosesogwo) [inline bubble_sort.rb]: This blank line and others within your code seem to be unnecessary.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @Mikearaya, @teekaytech:clap:. You have done well here:+1:, but we need to fix somethings**

See my comments on your code.


_**Kindly make those changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

File: bubble_sort.rb (hunk 1 above)
Comment (Mikearaya) [reply bubble_sort.rb]: **Noted**

File: bubble_sort.rb (hunk 2 above)
Comment (Mikearaya) [reply bubble_sort.rb]: **Noted**

File: bubble_sort.rb (hunk 3 above)
Comment (Mikearaya) [reply bubble_sort.rb]: **Noted** I put that there as a fail-safe :safety_pin:

File: bubble_sort.rb (hunk 4 above)
Comment (Mikearaya) [reply bubble_sort.rb]: **Noted**

**Comment by Mikearaya** [comment]
@mosesogwo Thank you for your review.
We have fixed all the issues outlined and are kindly requesting for another code review so we can merge to `master`

**Thank you**:bowing_man:

**Review by euqueme — APPROVED** [review]
> ### Pull request under the review by Maria Eugenia Queme P. (TSE) 

### Hello @Mikearaya and @teekaytech :balloon: :lollipop: 

### Thank you for doing the changes requested by the previous reviewer, everything works fine so you may move on congratulations :tada: :tada: :tada: 

[OPTIONAL]

- [x] favor the use of `array.size` instead of `array.lenght` to keep the ruby convention

PROJECT STATUS: APPROVED :green_apple:

**Pull request merged**